lib/floki/selector_parser.ex

```diff
@@ -42,49 +42,33 @@ defmodule Floki.SelectorParser do
     do_parse(t, %{selector | attributes: [result | selector.attributes]})
   end
   defp do_parse([{:pseudo_not, _} | t], selector) do
-    {t, pseudo_class} = do_parse_pseudo_not(t, %PseudoClass{name: "not"})
-
-    do_parse(t, %{selector | pseudo_class: pseudo_class})
+    {t, pseudo_not_class} = do_parse_pseudo_not(t, %PseudoClass{name: "not", value: []})
+    pseudo_classes = [pseudo_not_class | selector.pseudo_classes] |> Enum.reject(&is_nil(&1))
+    do_parse(t, %{selector | pseudo_classes: pseudo_classes})
   end
   defp do_parse([{:pseudo, _, pseudo_class} | t], selector) do
-    do_parse(t, %{selector | pseudo_class: %PseudoClass{name: to_string(pseudo_class)}})
+    pseudo_classes = [%PseudoClass{name: to_string(pseudo_class)} | selector.pseudo_classes]
+    do_parse(t, %{selector | pseudo_classes: pseudo_classes})
   end
   defp do_parse([{:pseudo_class_int, _, pseudo_class_int} | t], selector) do
-    pseudo_class = selector.pseudo_class
-    do_parse(t, %{selector | pseudo_class: %{pseudo_class | value: pseudo_class_int}})
+    [pseudo_class | pseudo_classes] = selector.pseudo_classes
+    do_parse(t, %{selector | pseudo_classes: [%{pseudo_class | value: pseudo_class_int} | pseudo_classes]})
   end
   defp do_parse([{:pseudo_class_even, _} | t], selector) do
-    pseudo_class = selector.pseudo_class
-    do_parse(t, %{selector | pseudo_class: %{pseudo_class | value: "even"}})
+    [pseudo_class | pseudo_classes] = selector.pseudo_classes
+    do_parse(t, %{selector | pseudo_classes: [%{pseudo_class | value: "even"} | pseudo_classes]})
   end
   defp do_parse([{:pseudo_class_odd, _} | t], selector) do
-    pseudo_class = selector.pseudo_class
-    do_parse(t, %{selector | pseudo_class: %{pseudo_class | value: "odd"}})
+    [pseudo_class | pseudo_classes] = selector.pseudo_classes
+    do_parse(t, %{selector | pseudo_classes: [%{pseudo_class | value: "odd"} | pseudo_classes]})
   end
   defp do_parse([{:pseudo_class_pattern, _, pattern} | t], selector) do
-    pseudo_class = selector.pseudo_class
-    do_parse(t, %{selector | pseudo_class: %{pseudo_class | value: to_string(pattern)}})
+    [pseudo_class | pseudo_classes] = selector.pseudo_classes
+    do_parse(t, %{selector | pseudo_classes: [%{pseudo_class | value: to_string(pattern)} | pseudo_classes]})
   end
   defp do_parse([{:pseudo_class_quoted, _, pattern} | t], selector) do
-    pseudo_class = selector.pseudo_class
-    do_parse(t, %{selector | pseudo_class: %{pseudo_class | value: to_string(pattern)}})
-  end
-  defp do_parse([{:pseudo_class_generic_value, _, value} | t], selector) do
```

Comment:
I couldn't find anywhere this was used and it seemed to be older, duplicate code so I went ahead and removed it in 9b2eee2. Let me know if it needs to stay in.